### PR TITLE
[eclipse/xtext#1855] update to junit 5.7

### PR DESCRIPTION
**Pull request by cdietrich**
[eclipse/xtext#1855] update to junit 5.7
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

File: releng/pom.xml [hunk 1]
```diff
@@ -12,7 +12,7 @@
 		<tycho-version>1.7.0</tycho-version>
 		<root-dir>${basedir}/..</root-dir>
 		<JENKINS_URL>https://ci.eclipse.org/xtext</JENKINS_URL>
-		<upstreamBranch>master</upstreamBranch>
+		<upstreamBranch>cd_junit57</upstreamBranch>
```
Comment (cdietrich) [inline releng/pom.xml:15]: to be undone with adjustPipelines

Comment (cdietrich) [reply releng/pom.xml:15]: arggg

File: org.eclipse.xtext.tests/testdata/wizard-expectations/mavenTychoJ11/mavenTychoJ11.parent/mavenTychoJ11.tests/META-INF/MANIFEST.MF [hunk 2]
```diff
@@ -6,7 +6,7 @@ Bundle-Version: 1.0.0.qualifier
 Bundle-SymbolicName: mavenTychoJ11.tests; singleton:=true
 Bundle-ActivationPolicy: lazy
 Require-Bundle: mavenTychoJ11,
- org.junit;bundle-version="4.12.0",
+ org.junit.jupiter.api;bundle-version="[5.0.0,6.0.0)",
```
Comment (kthoms) [inline org.eclipse.xtext.tests/testdata/wizard-expectations/mavenTychoJ11/mavenTychoJ11.parent/mavenTychoJ11.tests/META-INF/MANIFEST.MF:9]: The Oomph setup has a lower bound of 5.1. I think we had some reason with the JU extension. Just for correctness,  I'd like to see a 5.1 lower bound here. Practically it won't matter though.
(also locations below)

Comment (cdietrich) [reply org.eclipse.xtext.tests/testdata/wizard-expectations/mavenTychoJ11/mavenTychoJ11.parent/mavenTychoJ11.tests/META-INF/MANIFEST.MF:9]: done

**Review by kthoms — APPROVED** [review]

**Pull request merged**